external/vcm/vcm/_preloaded.py

```diff
@@ -0,0 +1,61 @@
+from typing import Sequence, Callable, Any
+import collections.abc
+import random
+import concurrent.futures
+
+
+class Preloaded(collections.abc.Iterator):
```

Comment:
Seems like Andre agrees: https://github.com/VulcanClimateModeling/fv3net/pull/1103/files#r604349872

Reply:
Yes, I see the motivation. This is a lightweight Python module that doesn't depend on anything in `loaders`, and it's not specific to loading data (it can also be used to pre-compute output). `vcm` is supposed to be the module to put lightweight python utility functions. My current scripts for this training use nothing from `loaders`, and I'd like to avoid introducing a dependency on that module's code.

My code does have a dependency on loaders through fv3fit, but only because the `__main__` script in fv3fit, and we've talked about how it would be ideal to remove that coupling.

Reply:
Can you just move it since Andre and I both agreed?

Reply:
It could fit in vcm sure, but it is highly related to loaders._batches.sequence.  It would decrease the cohesiveness of the code to split such related functionality. `fv3fit` depends on loaders.